checks/metrics.go

```diff
@@ -40,9 +41,20 @@ func getOrAddPrometheusMetric(name, metricType string, labelNames []string) (pro
 	return collector, prometheus.Register(collector)
 }
 
-func getWithEnvironment(ctx *context.Context, r *pkg.CheckResult) *context.Context {
+func getWithEnvironment(ctx *context.Context, r *pkg.CheckResult) (*context.Context, error) {
+	// Convert result Data into JSON for templating
+	// as gomplate does not render child map data correctly
+	var rData map[string]any
```

Comment:
Fixing this issue in https://github.com/flanksource/canary-checker/pull/1288 
I want to try and avoid unnecessary marshalling  - if it needs to happen should be in gomplate.